core/new-gui/src/app/workspace/component/code-editor-dialog/code-editor-dialog.component.ts

```diff
@@ -36,20 +37,21 @@ export class CodeEditorDialogComponent implements AfterViewInit, SafeStyle, OnDe
     fontSize: "11",
     automaticLayout: true,
   };
-  code: string;
   @ViewChild("editor", { static: true }) divEditor: ElementRef | undefined;
   loaded: boolean = false;
 
+  private formControl: FormControl;
   private ytext?: YText;
+  private editor?: any;
 
   constructor(
     private sanitizer: DomSanitizer,
     private dialogRef: MatDialogRef<CodeEditorDialogComponent>,
-    @Inject(MAT_DIALOG_DATA) code: any,
+    @Inject(MAT_DIALOG_DATA) formControl: FormControl,
     private workflowActionService: WorkflowActionService,
     public coeditorPresenceService: CoeditorPresenceService
   ) {
-    this.code = code;
+    this.formControl = formControl;
```

Comment:
ok.. I see you intention is to use formControl's status to change editor to be editable or not... but I think formControl should not be the source of truth for this behavior. We should have a workflow editable or not boolean somewhere, maybe in WorkflowActionService, here the editor should subscribe to that boolean's state, which is the single source of truth. What do you think?

Reply:
No we cannot because we allow temporarily enabling of only properties of an operator to debug the operator which is separate from the `checkWorkflowModificationEnabled()` in workflowActionService. The temporarily enabled status is only in the form component. I explained this in the description :)

Reply:
I did not fully understand that part from your description. No I have better understanding with https://github.com/Texera/texera/pull/1763#discussion_r1043768712. We can continue discussion there.